default/AI/ShipDesignAI.py

```diff
@@ -1924,6 +1910,77 @@ def _starting_guess(self, available_parts, num_slots):
             ret_val[-1] = num_slots - 1
         return ret_val
 
+
+def _add_shipdesign_fo(design_name, hull, parts_list, model="fighter", description="", icon="", name_desc_in_string_table=False, verbose=False):
+    """This is basically a wrapper around fo.issueCreateShipDesignOrder to also update the cache
+
+    :param design_name: the name of the design
+    :type design_name: str
+    :param description: a human readable description of the design
+    :type description: str
+    :param hull: the name of the hull to use
```

Comment:
Are you using some different version of the AI interface?

```int IssueCreateShipDesignOrder(const std::string& name, const std::string& description,
                                   const std::string& hull, const std::vector<std::string> parts,
                                   const std::string& icon, const std::string& model, bool nameDescInStringTable)``` 

It is indeed the hull name as string.